### PR TITLE
Prevent objects vanishing from view after manually resizing the client

### PR DESCRIPTION
**Pull request by sicoderealm**
Minor adjustment to FOV calculations to cater for all client window sizes.

Resolves: #463

### CODE REVIEW

**Review by akirmse — COMMENTED** [review]
I tried this and the FOV is way too small.  See screenshots.

Before:
before

After:
after

**Comment by akirmse** [comment]
Looking at this some more, I don't think this code is the source of the problem.  The FOV is correct for rendering, but wherever objects are tested for visibility there must be a bug that's still assuming the old fixed width and height values.

File: clientd3d/d3drender.c [hunk 1]
```diff
@@ -19,14 +19,15 @@ extern int main_viewport_width;
...
+inline float FovHorizontal(long width)
 {
-   return (diff / (float)(main_viewport_width) * (-PI / 3.6f));
+	return width / (float)(main_viewport_width) * (-PI / 3.6f);
 }
-inline float FovVertical(long diff)
+inline float FovVertical(long height)
 {
-   return (diff / (float)(main_viewport_height) * (PI / 5.6f));
+	return height / (float)(main_viewport_height) * (PI / 6.0f);
```
Comment (akirmse) [inline clientd3d/d3drender.c:28]: Should this 6.0 go back to 5.6?

File: clientd3d/draw3d.c [hunk 2]
```diff
@@ -401,7 +401,7 @@ void UpdateRoom3D(room_type *room, Draw3DParams *params)
    horizon = area.cy/2 + PlayerGetHeightOffset();
    num_visible_objects = 0;
    t1=timeGetTime();
-   DrawBSP(room, params, area.cx, FALSE);
+   DrawBSP(room, params, main_viewport_width, FALSE);
```
Comment (akirmse) [inline clientd3d/draw3d.c:404]: This is in the software renderer.  The bug is in the hardware renderer.

**Comment by davidklaw** [comment]
Can anyone explain the change here? Was the code previously only drawing objects within a limited range, but now does so for the entire viewport width? Is viewport height factored in or irrelevant, because I noticed that the vanishing object issues existed at all 4 viewport edges.

**Comment by davidklaw** [comment]
> solves this issues for all but the very smallest of sizes.

I'm testing at 2560x1440 (native) and at 1920x1080 and I only see this issue when the viewport is smaller than I would imagine anyone would ever want.

**Pull request merged**